### PR TITLE
add static MiddlewareFunc

### PR DESCRIPTION
**Pull request by wk989898**
- enable set Context with app.static
- add static example

### CODE REVIEW

**Comment by zhmushan** [comment]
@wk989898 see https://github.com/zhmushan/abc/blob/master/docs/static_files.md

**Comment by zhmushan** [comment]
You should run `deno fmt` befor commit

File: app.ts [hunk 1]
```diff
@@ -145,7 +145,7 @@ export class Application {
       for (const m of middleware) {
         h = m(h);
       }
-      return h(c);
+      return h(c) || handler(c);
```
Comment (zhmushan) [inline app.ts:148]: This is superfluous

File: app.ts [hunk 2]
```diff
@@ -158,15 +158,13 @@ export class Application {
...
+  static(prefix: string, root: string, ...m: MiddlewareFunc[]): Application {
     const h: HandlerFunc = (c) => {
       const filepath: string = c.params.filepath;
       return c.file(path.join(root, filepath));
     };
-    if (prefix === "/") {
-      return this.get(`${prefix}*filepath`, h);
-    }
-    return this.get(`${prefix}/*filepath`, h);
+    prefix = prefix.replace(/(\/)$/, "");
```
Comment (zhmushan) [inline app.ts:166]: 👍

File: app.ts (hunk 1 above)
Comment (wk989898) [reply app.ts:148]: if not ,i have to  write MiddlewareFunc with `return h(c)`
like this 
``` js
(h) =>
  (c) => {
    c.response.headers.set("Access-Control-Allow-Origin", "*");
    return h(c)  //  necessary
  },
```

**Pull request merged**